src/main/java/org/breedinginsight/api/auth/AuthServiceLoginHandler.java

```diff
@@ -75,10 +83,54 @@ public HttpResponse loginSuccess(UserDetails userDetails, HttpRequest<?> request
         return super.loginFailed(authenticationFailed);
     }
 
+    @Override
+    protected HttpResponse loginSuccessWithCookies(List<Cookie> cookies) {
+        try {
+            String locationUrl = this.jwtCookieConfiguration.getLoginSuccessTargetUrl();
+
+            Optional<HttpRequest<Object>> requestOptional = ServerRequestContext.currentRequest();
+            if (requestOptional.isPresent()){
+                HttpRequest<Object> request = requestOptional.get();
+                if (request.getCookies().contains(loginSuccessUrlCookieName)){
+                    Cookie loginSuccessCookie = request.getCookies().get(loginSuccessUrlCookieName);
+                    String returnUrl = loginSuccessCookie.getValue();
+                    try {
+                        returnUrl = URLDecoder.decode(returnUrl, StandardCharsets.UTF_8.name());
+                        if (isValidURL(returnUrl)){
+                            locationUrl = returnUrl;
+                        }
+                    } catch (UnsupportedEncodingException e){}
```

Comment:
May be useful to have a log entry instead of silent exception.

Reply:
changed and pushed